src/amg/azext_amg/migrate.py

```diff
@@ -0,0 +1,259 @@
+from .backup_core import get_all_dashboards, get_all_library_panels, get_all_snapshots, get_all_folders, get_all_annotations, get_all_datasources
+
+logger = get_logger(__name__)
+
+
+def migrate(backup_url, backup_headers, restore_url, restore_headers, dry_run, overwrite, folders_to_include=None, folders_to_exclude=None):
+    # get all datasources to be backedup
+    all_datasources = get_all_datasources(backup_url, backup_headers)
+    all_restore_datasources = get_all_datasources(restore_url, restore_headers)
+    if (all_restore_datasources is None) or (all_datasources is None):
+        logger.error("ABORTING!! Datasources are not found. Please check the URLs, headers, or api key/service token. Try running with --verbose.")
+        return
+    (datasources_created_summary, datasources_remapped_summary) = _migrate_datasources(all_datasources, all_restore_datasources, restore_url, restore_headers, dry_run)
+
+    all_folders = get_all_folders(backup_url, backup_headers, folders_to_include=folders_to_include, folders_to_exclude=folders_to_exclude)
+    all_restore_folders = get_all_folders(restore_url, restore_headers, folders_to_include=folders_to_include, folders_to_exclude=folders_to_exclude)
+    (folders_created_summary, folders_overwrote_summary) = _migrate_folders(all_folders, all_restore_folders, restore_url, restore_headers, dry_run, overwrite)
+
+    valid_folder_uids = set(folder[0]['uid'] for folder in all_folders + all_restore_folders)
+    all_dashboards = get_all_dashboards(backup_url, backup_headers, folders_to_include=folders_to_include, folders_to_exclude=folders_to_exclude)
+    all_library_panels = get_all_library_panels(backup_url, backup_headers)
+    if folders_to_exclude is None or 'General' not in folders_to_exclude:
+        # edge case for general folder. The uid is empty string.
+        valid_folder_uids.add('')
+
+    all_library_panels_filtered = [panel for panel in all_library_panels if panel['folderUid'] in valid_folder_uids]
+    (library_panels_created_summary, library_panels_overwrote_summary, dashboards_created_summary, dashboards_overwrote_summary) = _migrate_library_panels_and_dashboards(all_dashboards, all_library_panels_filtered, restore_url, restore_headers, dry_run, overwrite)
+
+    # get the list of snapshots to backup
+    all_snapshots = get_all_snapshots(backup_url, backup_headers)
+    snapshots_synced_summary = _migrate_snapshots(all_snapshots, restore_url, restore_headers, dry_run)
+   
+    # get all the annotations
+    all_annotations = get_all_annotations(backup_url, backup_headers)
+    annotations_synced_summary = _migrate_annotations(all_annotations, restore_url, restore_headers, dry_run)
+
+    dry_run_status = "to be " if dry_run else ""
+    output = [
+        (Style.IMPORTANT, f"\n\nDry run: {dry_run if dry_run else False}\n"),
+        (Style.IMPORTANT, f"Overwrite dashboards, folders, and library panels: {overwrite if overwrite else False}\n"),
```

Comment:
dry_run and overwrite are booleans right? Can these both just be simplified to {dry_run} and {overwrite} ?